src/Sentry.AspNetCore/SentryMiddleware.cs

```diff
@@ -131,16 +131,16 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
                 // when the event fires.  Use `activeScope`, not `scope` or `hub`.
                 scope.OnEvaluating += (_, activeScope) =>
```

Comment:
The outer `ConfigureScope` call no longer allocates a closure, but this `OnEvaluating` event handler still allocates unconditionally. One solution might be to store `KeyValuePair`s of the callback and its argument instead of using an event inside a `Stack`. Perhaps the `Stack` wouldn't even be necessary when the `Sentry.AspNetCore.Grpc` integration isn't loaded, as that seems to be the only other place (outside of tests) where this event handler is assigned.

Reply:
Definitely seems best to leave this to a separate PR.

Is there any infrastructure in place to measure the overhead of `SentryMiddleware`/`SentryTracingMiddleware` both in terms of time and memory? If not, that would  probably be a good thing to have. As these middleware implementations are invoked for every request if the SDK is enabled, it makes sense to optimize them as much as possible IMO.

Reply:
Sounds good. I've created a separate issue to track this:
- https://github.com/getsentry/sentry-dotnet/issues/4253